improved logs readability and config

sv3ndk opened the pull request:
- passing --logging-level=DEBUG  at the command line is no longer ignored
- classes can now implement `WithLogger` to get access to convenience logging function (e.g. `self.log_info("blah"))`, producing log statement with a logger whose name is the name of the class. The resulting log statements look like this:

```
2015-12-07 17:22:43,084 - BackfillJob - INFO - [backfill progress] waiting: 0 | succeeded: 1 | kicked_off: 1 | failed: 0 | wont_run: 0
2015-12-07 17:22:43,084 - SequentialExecutor - DEBUG - Calling the <class 'airflow.executors.sequential_executor.SequentialExecutor'> sync method
2015-12-07 17:22:43,084 - SequentialExecutor - DEBUG - 0 running task instances
2015-12-07 17:22:43,084 - SequentialExecutor - DEBUG - 0 in queue
2015-12-07 17:22:43,084 - SequentialExecutor - DEBUG - 32 open slots
2015-12-07 17:22:43,085 - BackfillJob - INFO - All done. Exiting.
```
- configured this WithLogger super class in a couple of key existing classes to improve current log readability.

sv3ndk committed:
improved logs readability and config

- passing --logging-level=DEBUG  at the command line is no longer ignored
- classes can now implement WithLogger to get access to convenience logging function (e.g. self.log_info("blah")), producing log statement with a logger whose name is the name of the class

e.g.:

2015-12-07 17:22:43,084 - BackfillJob - INFO - [backfill progress] waiting: 0 | succeeded: 1 | kicked_off: 1 | failed: 0 | wont_run: 0
2015-12-07 17:22:43,084 - SequentialExecutor - DEBUG - Calling the <class 'airflow.executors.sequential_executor.SequentialExecutor'> sync method
2015-12-07 17:22:43,084 - SequentialExecutor - DEBUG - 0 running task instances
2015-12-07 17:22:43,084 - SequentialExecutor - DEBUG - 0 in queue
2015-12-07 17:22:43,084 - SequentialExecutor - DEBUG - 32 open slots
2015-12-07 17:22:43,085 - BackfillJob - INFO - All done. Exiting.

- configured this WithLogger super class in a couple of key classes to improve log readability

sv3ndk committed:
add 2 UT for log_to_stdout()

sv3ndk committed:
added one more UT for WIthLogger + a few more tested cases +  minor code cleanup

mistercrunch commented on airflow/settings.py:
```diff
@@ -43,6 +42,48 @@
 SIMPLE_LOG_FORMAT = '%(asctime)s %(levelname)s - %(message)s'
 
 
+def log_to_stdout():
```
Can we move this and WithLogger to `airlfow.utils` instead of settings?

mistercrunch commented on airflow/settings.py:
```diff
@@ -43,6 +42,48 @@
+    root_logger.addHandler(ch)
+
+
+class WithLogger(object):
```
`LogMixin`, `LoggingMixin`?

mistercrunch commented on airflow/settings.py:
```diff
@@ -43,6 +42,48 @@
+    """
+
+    def logger(self):
+        return logging.getLogger(self.__class__.__name__)
```
Should the logger object be memoized somehow, otherwise getLogger is called at every logging event...

sv3ndk committed:
adapts name and behaviour of WithLogger

- logger is now accessed as a property
- logger is now memoized
- WithLogger is renamed to LoggingMixin + moved to utils.py
- mentioned the --logging-level=DEBUG option in CONTRIB.md

sv3ndk committed:
add UT for round_time()

(in order to make coverage checks ok ^__^)

sv3ndk committed:
adding one UT for log_to_stdout
